### PR TITLE
Upgrade trussworks/react-uswds v9 and uswds v3.8

### LINKED ISSUE #7537: Upgrade Trussworks to version 9
### Background
Our Trussworks implementation is on 5.1.1 and should be upgrade closer to current active version 9.0.0. The older version is causing [one of our dependabot upgrades to break](https://github.com/CDCgov/prime-simplereport/pull/7528?notification_referrer_id=NT_kwDOAcRY8LQxMDE2MTk4MDg5OToyOTY0NTA0MA#issuecomment-2042771646) 

### Action requested
Upgrade Truss / associated peer dependencies
Fix any frontend UI / testing issues that come up

### Acceptance Criteria
Truss is upgraded to latest version 

### Additional context
Initial attempt that I'm punting to whoever picks this up is [here](https://github.com/CDCgov/prime-simplereport/pull/7536)

### PR DESCRIPTION
**Pull request by emyl3**
# FRONTEND PULL REQUEST

- Resolves #7537

## Changes Proposed

- Upgrades `@trussworks/react-uswds` to `9.0.0` and `@uswds/uswds` to `3.8.0`
- Added a dev dependency [`focus-trap-react`](https://github.com/trussworks/react-uswds/pull/2820) that is now required by `@trussworks/react-uswds`.

## Additional Information
As discussed [here](https://skylight-hq.slack.com/archives/C065V4DHBQF/p1714673660474579), the disabled state styling has changed.

**Before**

Screenshot 2024-05-02 at 13 12 03

**After**

Screenshot 2024-05-02 at 13 01 29

## Testing
- Smoke test parts of the app that are using [components](https://github.com/search?q=repo%3ACDCgov%2Fprime-simplereport+%40trussworks%2Freact-uswds+language%3ATSX&type=code&l=TSX) from `@trussworks/react-uswds`
- Deployed on dev2

### CODE REVIEW

**Review by fzhao99 — APPROVED** [review]
💪

**Review by mpbrown — APPROVED** [review]
LGTM! Nice work on resolving this upgrade!

**Comment by emyl3** [comment]
@fzhao99 @mpbrown Just noticed a change in the button styles causing some buttons with icons to look different 😓 Sorry about that! Will fix that up and re-request review!

**Review by bobbywells52 — DISMISSED** [review]

**Comment by emyl3** [comment]
@fzhao99 @mpbrown @bobbywells52 Ready for re-review (Sorry for making you review it twice) and is being deployed on dev6!

**Comment by emyl3** [comment]
As requested by @fzhao99, (the "after" images are after the fix went in place to set the `column-gap` prop to `0`).

**Before - tool tipicon**
Screenshot 2024-05-07 at 10 12 49

**After - tooltip icon**
Screenshot 2024-05-07 at 10 12 40

**Before - language toggle**
Screenshot 2024-05-07 at 10 14 27

**After - language toggle**
Screenshot 2024-05-07 at 10 14 50

**Before - add another button**
Screenshot 2024-05-07 at 10 18 16

**After - add another button**
Screenshot 2024-05-07 at 10 16 53

**Review by mpbrown — COMMENTED** [review]
I noticed the disabled state for the checkboxes on the facility access component don't necessarily seem like they make it clear that these mean they are selected (comparing these two screenshots below) Are we good on the new design for this or should be override this Trussworks/USWDS change?

image

image

**Comment by mpbrown** [comment]
> As requested by @fzhao99, (the "after" images are after the fix went in place to set the `column-gap` prop to `0`).

@emyl3 not sure if I'm just noticing it but could you highlight what you mean by the differences in those screenshots?

**Comment by emyl3** [comment]
> I noticed the disabled state for the checkboxes on the facility access component don't necessarily seem like they make it clear that these mean they are selected (comparing these two screenshots below) Are we good on the new design for this or should be override this Trussworks/USWDS change?
> 
> image
> 
> image

OOO nice catch -- will investigate this!

**Comment by emyl3** [comment]
> > As requested by @fzhao99, (the "after" images are after the fix went in place to set the `column-gap` prop to `0`).
> 
> @emyl3 not sure if I'm just noticing it but could you highlight what you mean by the differences in those screenshots?

There shouldn't be much of a difference the "after" images. They should look nearly the same since I had to add a class to buttons with icons since the buttons for the upgraded version introduced a new css property of `column-gap` set to `.5rem`. With the css class, I set that `column-gap` to `0` to remove the extra spacing between the icon and button text. Hope this makes sense 😅

File: frontend/src/scss/base/_prime-styles.scss [hunk 1]
```diff
@@ -1133,13 +1133,6 @@ $results-dropdown-spacing: #{units(4)} - #{units(2)} - 22px - #{units(4)}; // he
   border: 2px solid #dfe1e2;
 }
 
-.usa-checkbox__input:disabled + .usa-checkbox__label::before,
```
Comment (emyl3) [inline frontend/src/scss/base/_prime-styles.scss:1136]: @mpbrown Removing this should have fixed it! Thank you for catching that!
screenshot of dev6
Screenshot 2024-05-07 at 11 25 20

**Review by fzhao99 — APPROVED** [review]
lgtm! Thanks for your work on this :)

**Review by mpbrown — APPROVED** [review]

**Pull request merged**